### PR TITLE
all Endpoints updated including running tests

### PR DESCRIPTION
**Pull request by samBAM089**
behause i had too many changes I had to close my previous PRs.

### CODE REVIEW

File: backend/src/main/java/de/sambam/gamenighttracker/controller/UserController.java [hunk 1]
```diff
@@ -24,10 +21,24 @@ public UserController(UserService userService) {
         this.userService = userService;
     }
 
-    @GetMapping({"/games"})
-    public List<Game> gameList() {
-        return userService.listAllGames("1")
-                .orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST, "User not found"));
+    @GetMapping("/games")
+    public List<Game> getGameList() {
+        return userService.listAllGames("1");
```
Comment (jamarob) [inline backend/src/main/java/de/sambam/gamenighttracker/controller/UserController.java:26]: You want to later get the player id from the principal, I guess?

File: backend/src/main/java/de/sambam/gamenighttracker/service/UserService.java [hunk 2]
```diff
@@ -19,11 +23,70 @@ public UserService(UserDb userDb) {
         this.userDb = userDb;
     }
 
-    public Optional<List<Game>> listAllGames(String id) {
+    public List<Game> listAllGames(String id) {
```
Comment (jamarob) [inline backend/src/main/java/de/sambam/gamenighttracker/service/UserService.java:26]: I think the naming can be improved. Maybe something like ´listAllGameByPlayer´ and/or rename `id` to `userId`?

File: backend/src/test/java/de/sambam/gamenighttracker/controller/UserControllerTest.java [hunk 3]
```diff
@@ -35,46 +36,172 @@
 
     @BeforeEach
     public void setup() {
+        Player player1 = Player.builder().name("Mario").color("red").score(123)
```
Comment (jamarob) [inline backend/src/test/java/de/sambam/gamenighttracker/controller/UserControllerTest.java:39]: Maybe extract the creation of games for testing into helper methods, helps with readability.

**Review by jamarob — APPROVED** [review]
Good job! 🎲 I think your tests could be improved in readability by introducing a factory object for some stock players and games.

File: backend/src/main/java/de/sambam/gamenighttracker/controller/UserController.java (hunk 1 above)
Comment (samBAM089) [reply backend/src/main/java/de/sambam/gamenighttracker/controller/UserController.java:26]: probably ^^'

File: backend/src/main/java/de/sambam/gamenighttracker/controller/UserController.java [hunk 4]
```diff
@@ -24,10 +21,24 @@ public UserController(UserService userService) {
...
+        return userService.listAllPlayers("1");
+    }
+
+    @PostMapping("/game")
+    public Optional<Game> addNewGame(@RequestBody Game gameToAdd) {
+        return userService.addNewGame(gameToAdd, "1");
+    }
+
+    @PostMapping("/game/{apiGameId}/gamesessions")
+    public Optional<GameSession> addNewGameSession(@RequestBody GameSession newSession, @PathVariable("apiGameId") String apiGameId) {
```
[the remark this answers is not in the document]
Comment (samBAM089) [reply backend/src/main/java/de/sambam/gamenighttracker/controller/UserController.java:44]: thanks ^^

File: backend/src/test/java/de/sambam/gamenighttracker/controller/UserControllerTest.java (hunk 3 above)
Comment (samBAM089) [reply backend/src/test/java/de/sambam/gamenighttracker/controller/UserControllerTest.java:39]: I save that for the last 2 days of the project phase ^^

**Pull request merged**